pkg/heatcfnapi/deployment.go

```diff
@@ -136,7 +139,7 @@ func Deployment(
 	deployment.Spec.Template.Spec.Affinity = affinity.DistributePods(
 		common.AppSelector,
 		[]string{
-			heat.ServiceName,
+			labels[common.AppSelector],
```

Comment:
I agree that we had a problem with all pods receiving the `api` component label. That has been fixed by the other change. But maybe we should still use two labels to preserve the common `service` grouping of pods?

Reply:
This ultimately gives the user more flexibility when trying to filter the pods their after:
```
❯ oc get po -l service=heat,component=engine
NAME                           READY   STATUS    RESTARTS   AGE
heat-engine-7fb9fb9c4d-xhf6t   1/1     Running   0          20s
```

Which would be more important in scenarios where they have multiple replicas of a deployment like `heat-engine` for example:

```
❯ oc get po -l service=heat,component=engine
NAME                           READY   STATUS     RESTARTS   AGE
heat-engine-7fb9fb9c4d-68jcr   0/1     Init:0/1   0          1s
heat-engine-7fb9fb9c4d-6p9tz   0/1     Init:0/1   0          1s
heat-engine-7fb9fb9c4d-kglc8   0/1     Init:0/1   0          1s
heat-engine-7fb9fb9c4d-xhf6t   1/1     Running    0          109s
heat-engine-7fb9fb9c4d-z66ct   0/1     Init:0/1   0          1s
```